File: python/day-10.py
#!/bin/python3

import math
import os
import random
import re
import sys



if __name__ == '__main__':
    n = int(input())
    binary_value = str(bin(n)[2:])
    consecutive_counter = 0
    max_consecutive_counter = 0

    for c in binary_value:
        if c == str(1):
            consecutive_counter = consecutive_counter + 1
            if max_consecutive_counter < consecutive_counter:
                max_consecutive_counter = consecutive_counter
        else:
            if max_consecutive_counter < consecutive_counter:
                max_consecutive_counter = consecutive_counter
                
            consecutive_counter = 0
            
    print(max_consecutive_counter)

File: java/day-10.java
import java.io.*;
import java.math.*;
import java.security.*;
import java.text.*;
import java.util.*;
import java.util.concurrent.*;
import java.util.regex.*;

public class Solution {

    private static final Scanner scanner = new Scanner(System.in);

    public static void main(String[] args) {
        int n = scanner.nextInt();
        scanner.skip("(\r\n|[\n\r\u2028\u2029\u0085])?");

        String binaryValue = Integer.toBinaryString(n);
        int consecutiveCounter = 0;
        int maxCC = 0;
        
        for (char c: binaryValue.toCharArray()) {
            
            if (c == '1') {
                consecutiveCounter = consecutiveCounter + 1;
                if (maxCC < consecutiveCounter) {
                    maxCC = consecutiveCounter;
                }
            }
            else {
                if (maxCC < consecutiveCounter) {
                    maxCC = consecutiveCounter;
                }
                consecutiveCounter = 0;
            }
        
        }
        
        System.out.println(maxCC);
        scanner.close();
    }
}
